rag_experiment_accelerator/run/index.py

```diff
@@ -42,68 +112,432 @@ def run(config_dir: str, data_dir: str = "data", filename: str = "config.json")
-                        )
-                        index_dict["indexes"].append(index_name)
 
-    index_output_file = f"{config.artifacts_dir}/generated_index_names.jsonl"
-    with open(index_output_file, "w") as index_name:
-        json.dump(index_dict, index_name, indent=4)
+def chunk_size_overlap_combinations(config: Config):
+    """
+    Generates combinations for different chunk sizes and overlaps defined
+    in the configuration
+
+    Args:
+        config (object): Configuration for rag accelerator
+
+    Returns:
+        A list of tuples for each chunk size and overlap combination
+    """
+
+    return itertools.product(config.CHUNK_SIZES, config.OVERLAP_SIZES)
+
+
+def create_search_index(config, service_endpoint, key, index_dict, chunk_size, overlap):
+    """
+    Creates a search index in Azure Search.
+
+    This function constructs the index name based on various configuration
+    parameters and then creates the index using the `create_acs_index`
+    function.
+    It logs the start and end of the index creation process and the time took.
+    The created index name is then added to the `index_dict` dictionary.
+
+    Args:
+        config (object): A configuration object that holds various settings.
+        service_endpoint (str): The Azure Search service endpoint.
+        key (str): The Azure Search service key.
+        index_dict (dict): A dictionary to store the created index names.
+        chunk_size (int): The size of the chunks of content.
+        overlap (int): The overlap size between chunks.
+        dimension (int): The size of the embedding vector.
+        ef_construction (int): The size of the dynamic list used during the
+        construction of the HNSW graph.
+        ef_search (int): The size of the dynamic list used during the search
+        phase.
+    """
+
+    for embedding_model in config.embedding_models:
+        for ef_construction in config.EF_CONSTRUCTIONS:
+            for ef_search in config.EF_SEARCHES:
+                index_name = generate_index_name(
+                    config,
+                    chunk_size,
+                    overlap,
+                    embedding_model,
+                    ef_construction,
+                    ef_search,
+                )
+                with TimeTook(f"create Azure Search Index {index_name}", logger=logger):
+                    create_acs_index(
+                        service_endpoint,
+                        index_name,
+                        key,
+                        embedding_model.dimension,
+                        ef_construction,
+                        ef_search,
+                        config.LANGUAGE["analyzers"],
+                    )
+                index_dict["indexes"].append(index_name)
+
+
+def generate_index_name(
+    config, chunk_size, overlap, embedding_model, ef_construction, ef_search
+):
+    """
+    Generates an index name considering config parameters
+
+    Args:
```

Comment:
These args don't appear to match the function signature.
e.g `dimension` below
Is this a CoPilot hallucination?

Reply:
Those parameters got changed overtime, and I forgot to update the docs. thanks!

Reply:
This one is still an issue

Reply:
same here:
![image](https://github.com/microsoft/rag-experiment-accelerator/assets/3179107/92ec28b0-28b2-4f72-b04e-d839addc6af4)